packages/core/src/types/state.ts

```diff
@@ -25,6 +25,7 @@ export interface State {
   canvas: Canvas;
   computeShapes: ShapeFn;
   params: Params;
+  frozenValues: number[];
```

Comment:
Could you add a comment about what this means? Are these indices into the `varyingValues` array?